### ISSUE TITLE
New Kindle 10 HD

### ISSUE THREAD

**Issue by JoshuaFern** [M0Rf30/android-udev-rules#256]
Hi, I ordered a Kindle around December and it has a different product id, could you add it to the list?

```
  ~ ❯ adb devices -l
List of devices attached
G8S1KT061474031B       device usb:3-4 product:trona model:KFTRWI device:trona transport_id:2

  ~ ❯ lsusb         
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 009: ID 1949:05e8 Lab126, Inc. Fire
Bus 003 Device 003: ID 0810:0001 Personal Communication Systems, Inc. Dual PSX Adaptor
Bus 003 Device 002: ID 1209:2303 Generic Atreus
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 004: ID 0a12:0001 Cambridge Silicon Radio, Ltd Bluetooth Dongle (HCI mode)
Bus 001 Device 003: ID 046d:c24a Logitech, Inc. G600 Gaming Mouse
Bus 001 Device 002: ID b58e:9e84 Blue Microphones Yeti Stereo Microphone
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

`ATTRS{idVendor}=="1949", ATTRS{idProduct}=="05e8"`

**Comment by JoesCat** [comment]
Hi @JoshuaFern,
Can you please try [kindle10 patch](https://github.com/JoesCat/android-udev-rules/tree/kindle10).

I think earlier versions of Kindle simply ran in debug mode, but searching around, I see various numbers when I look for Kindle10.
I am guessing that default mode with MTP access is 05e1 based on seeing this requested patch [#1913 Amazon Kindle fire 10 plus problem](https://sourceforge.net/p/libmtp/bugs/1913/), and the value you have 05e8 is for adb access?

Please test the patched 51-android-rules and verify it works.
Please try the various Kindle modes and let us know which numbers you see for the different modes (example, default, adb-debug, etc....).

This might help, even if it is for a different kindle - see pdf listed below:
https://stackoverflow.com/questions/8257201/how-can-usb-debugging-for-kindle-fire-can-be-enabled-on-a-linux-ubuntu
http://g-ecx.images-amazon.com/images/G/01/sdk/Connecting_your_Kindle_Fire_to_ADB.pdf

**Comment by M0Rf30** [comment]
Hi @JoshuaFern can you give feedback on testing this PR?
https://github.com/M0Rf30/android-udev-rules/pull/257
Thanks

**Comment by JoshuaFern** [comment]
Hi,
I haven't tested the PR yet as I've been busy, but I had manually added the udev rules with this product ID and vendor ID and it works.

> Please try the various Kindle modes and let us know which numbers you see for the different modes (example, default, adb-debug, etc....).
Could you give me specific instructions for how to enter each mode you'd like me to test? It's not clear how many modes there are or how to get into each of them.

**Comment by M0Rf30** [comment]
Generally speaking you have to itereate between different modes and cattch the vendor id /product id of an lsusb output

**Comment by JoshuaFern** [comment]
I'm not sure if these are all the modes but I went through and got the IDs for this device.

Fastboot: `Bus 001 Device 006: ID 1949:05e0 Lab126, Inc. Android`
MTP: `Bus 001 Device 011: ID 1949:05e1 Lab126, Inc. Fire`
MIDI: `Bus 001 Device 012: ID 1949:2046 Lab126, Inc. Fire`
PTP: `Bus 001 Device 013: ID 1949:05e3 Lab126, Inc. Fire`
ADB: `Bus 003 Device 009: ID 1949:05e8 Lab126, Inc. Fire`

**Comment by JoshuaFern** [comment]
I had some extra time so I got some IDs for another Android device, the Anbernic RG353P.

Fastboot: `Bus 001 Device 024: ID 18d1:d00d Google Inc. Xiaomi Mi/Redmi 2 (fastboot)`
MTP: `Bus 001 Device 018: ID 2207:0017 Fuzhou Rockchip Electronics Company RG353P`
USB Tethering: `Bus 001 Device 019: ID 2207:0013 Fuzhou Rockchip Electronics Company RG353P`
PTP: `Bus 001 Device 020: ID 2207:0018 Fuzhou Rockchip Electronics Company RG353P`
ADB: `Bus 001 Device 021: ID 2207:0006 Fuzhou Rockchip Electronics Company RG353P`

**Comment by JoesCat** [comment]
Sounds like you are on a roll.
Go for it!

:-)

**Comment by JoshuaFern** [comment]
PinePhone v1.2 running Glodroid
Fastboot: `Bus 001 Device 048: ID 18d1:4ee0 Google Inc. Nexus/Pixel Device (fastboot)`
MTP: `Bus 001 Device 040: ID 18d1:4ee2 Google Inc. Nexus/Pixel Device (MTP + debug)`
USB Tethering: `Bus 001 Device 041: ID 18d1:4ee4 Google Inc. Nexus/Pixel Device (tether+ debug)`
PTP: `Bus 001 Device 042: ID 18d1:4ee6 Google Inc. Nexus/Pixel Device (PTP + debug)`
ADB: `Bus 001 Device 036: ID 18d1:4ee7 Google Inc. Nexus/Pixel Device (charging + debug)`

It just occurred to me that I might have my terminology wrong, when I list an output as "adb" I mean charging + debug like above, which makes me wonder. Is there another device ID for MTP without debugging? Should I gather those as well?

Amazon Fire HD 8 2016 (giza)
Fastboot: `Bus 001 Device 061: ID 0bb4:0c01 HTC (High Tech Computer Corp.) Dream / ADP1 / G1 / Magic / Tattoo / FP1`
MTP: `Bus 001 Device 051: ID 1949:0232 Lab126, Inc. Fire`
PTP: `Bus 001 Device 052: ID 1949:0234 Lab126, Inc. Fire`

This version of FireOS has only MTP and PTP as selectable options in the USB screen.

**Comment by JoesCat** [comment]
Yes, when debug is turned-off there is also MTP, PTP,midi,charging......
A large percentage of users are not developers, so the debug-off mode is useful for adding to other rules like 69-mtp, libmtp and other libraries.
If you can clarify which are with debug enabled, or disabled, that is useful info.
If you look at some of the comments in 51-android-rules, you'll notice some have detailed info, like the Fairphone, while other phones lack a lot of info.
The info you have for the PinePhone looks great (mtp+debug....etc)
I guess the FireOS MTP,PTP are with debug off? on? both?

**Comment by JoesCat** [comment]
Looks good, but I see some conflicting info:

may be a copy-n-paste error == 05e1
Amazon Kindle 10 2021 (trona) - 7.3.2.8
MTP: Bus 001 Device 049: ID 1949:05e1 Lab126, Inc. Fire
No data transfer: Bus 001 Device 052: ID 1949:05e1 Lab126, Inc. Fire (charging)

Anbernic RG353P
MTP: Bus 001 Device 015: ID 2207:0007 Fuzhou Rockchip Electronics Company Pixel 6
No data transfer: Bus 001 Device 012: ID 2207:0007 Fuzhou Rockchip Electronics Company Pixel 6

...and this conflicts with the HTC MTP, and there are many users vs developers, so this could be a comment, but not activated (this avoids causing problems for the majority of users).
Amazon Kindle 8 2016 (giza) - 5.7.0.0
Fastboot: Bus 001 Device 061: ID 0bb4:0c01 HTC (High Tech Computer Corp.)

**Comment by JoshuaFern** [comment]
It's not a copy paste error, the IDs are the same in both modes - not sure why.

**Comment by JoesCat** [comment]
Maybe it could be (smart) charging, since it is connected to a USB socket.
Without details, it would charge at USB 1.0 rates, not fast charge.

I've added something to try-out, based on looking at other codes, particularly the Samsung mtp.
Give a try for:
adbmtp - might end up being some variant of something like this https://support.garmin.com/en-US/?faq=ycfanFPMus028WBG13MEOA
adbptp - no special extras added
adbmidi - you should see /dev/midi and /dev/midi0? devices - might need more.

I did not add plain mtp, or ptp, since this might be something to add to rules 69
plain midi, don't think is served by anything yet, maybe needs to be retained with 51-rules. ....just thinking out loud.....
